appengine_flexible/websockets/main_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/websocket"
+)
+
+func TestSocketHandler(t *testing.T) {
+	var err error
+
+	server := httptest.NewServer(http.HandlerFunc(socketHandler))
+	defer server.Close()
+	dialer := websocket.Dialer{}
+
+	conn, resp, err := dialer.Dial("ws://"+server.Listener.Addr().String()+"/ws", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	got := resp.StatusCode
+	want := http.StatusSwitchingProtocols
+	if got != want {
+		t.Errorf("resp.StatusCode = %q, want %q", got, want)
+	}
+
+	err = conn.WriteMessage(websocket.TextMessage, []byte("echo test"))
+	if err != nil {
```

Comment:
Does this confirm a message was sent back? I'm not sure under what conditions `WriteMessage` returns an error.

Reply:
This will fail if for some reason the websocket could not be written to. I have added an additional test which checks on the echo response.